pkg/scheduler/schedule_one.go

```diff
@@ -239,9 +239,13 @@ func (sched *Scheduler) bindingCycle(ctx context.Context, state *framework.Cycle
 			// Avoid moving the assumed Pod itself as it's always Unschedulable.
 			// It's intentional to "defer" this operation; otherwise MoveAllToActiveOrBackoffQueue() would
 			// update `q.moveRequest` and thus move the assumed pod to backoffQ anyways.
-			defer sched.SchedulingQueue.MoveAllToActiveOrBackoffQueue(internalqueue.AssignedPodDelete, func(pod *v1.Pod) bool {
-				return assumedPod.UID != pod.UID
-			})
+			if waitOnPermitStatus.IsUnschedulable() {
```

Comment:
why this change?

Reply:
For unschedulable pods, since it's useless to put it back to the backingoffQ/activeQ at that moment, so we should sent it to the `unschedulingPods`. Considering `MoveAllToActiveOrBackoffQueue()` would update `q.moveRequest` then `FailureHandler` will move the pod to backoffQ anyways, so we defer the move on purpose.

Reply:
So, on the other hand, if there is a failure, you want the pod to be back into backoff/active immediately, right?

@Huang-Wei wdyt?

Reply:
This is what `PreBind` and `Bind` phases behave like now. And WaitOnPermit here will always return a Unschedulable status when rejected, see https://github.com/kubernetes/kubernetes/blob/c870f1ddc73903c074b441a11fe16dff23fc1438/pkg/scheduler/framework/runtime/waiting_pods_map.go#L152-L165

I think we can unify the behaviors when getting different status, this is also what I want to achieve in this PR. Like 
- unschedulable -> unschedulablePods
- error -> backoffQ

Do we have any special cases ?

Reply:
> unschedulable -> unschedulablePods
> error -> backoffQ

The semantics LGTM.

It's just w/ this change, we start to honor the status returned by PreBind/Bind. For waitForPermit, a bit over-engineering though as it's not a public interface and in all cases it's treated as Unschedulable.

@cofyc FYI: for now, VolumeBind plugin's PreBind returns either Error or Success. In terms of Error, actually scheduler will retry them, is that intentional? If not, say you have already registered  a cluster event in EventsToRegister(), we can change Error to Unschedulable, so scheduler will hold rescheduling it until a related event is fired.

Reply:
> It's just w/ this change, we start to honor the status returned by PreBind/Bind. For waitForPermit, a bit over-engineering though as it's not a public interface and in all cases it's treated as Unschedulable.

The intention here is to unify the logic then in this PR https://github.com/kubernetes/kubernetes/pull/112025, I can extract them and reduce code redundant.

Reply:
While the semantics make sense, the nesting of the logic here is a future mine field; consider simplifying this somehow, perhaps converting this into a switch statement with cases?

Reply:
I think that's the intention. @kerthcet mentioned in the description section, that he will leave the real refactoring in the follow-up.

Reply:
Yeah, I'll leave the refactor to that PR.